runtime/src/snapshot_utils.rs

```diff
@@ -965,6 +966,19 @@ where
 
     let unpacked_version_file = unpack_dir.path().join("version");
     let snapshot_version = {
+        // Check size of snapshot_version file.
+        let file_size = fs::metadata(&unpacked_version_file)?.len();
+        if file_size > MAX_SNAPSHOT_VERSION_FILE_SIZE {
+            let error_message = format!(
+                "snapshot version file too large: {} has {} bytes (max size is {} bytes)",
+                unpacked_version_file.display(),
+                file_size,
+                MAX_SNAPSHOT_VERSION_FILE_SIZE,
+            );
+            return Err(get_io_error(&error_message));
+        }
+
+        // Read snapshot_version from file.
```

Comment:
Please move this into its own function (and all the way down to line 986, the end of this block, but I cannot select that far in GitHub...).

Named something like `snapshot_version_from_file()` and have it take an `impl AsRef<Path>` and return `Result<String>`.